platform/mbed_assert.h

```diff
@@ -77,61 +77,47 @@ do {                                                     \
  *  The assertion acts as a declaration that can be placed at file scope, in a
  *  code block (except after a label), or as a member of a C++ class/struct/union.
  *
- *  @note
- *  Use of MBED_STATIC_ASSERT as a member of a struct/union is limited:
- *  - In C++, MBED_STATIC_ASSERT is valid in class/struct/union scope.
- *  - In C, MBED_STATIC_ASSERT is not valid in struct/union scope, and
- *    MBED_STRUCT_STATIC_ASSERT is provided as an alternative that is valid
- *    in C and C++ class/struct/union scope.
- *
  *  @code
- *  MBED_STATIC_ASSERT(MBED_LIBRARY_VERSION >= 120,
- *          "The mbed library must be at least version 120");
+ *  MBED_STATIC_ASSERT(MBED_MAJOR_VERSION  >= 6,
+ *          "The mbed-os library must be at least version 6.0.0");
  *
  *  int main() {
  *      MBED_STATIC_ASSERT(sizeof(int) >= sizeof(char),
  *              "An int must be larger than a char");
  *  }
  *  @endcode
+ *
+ *  @deprecated This feature is now no longer necessary with the minimum
+ *  supported language versions. It will be removed in a forthcoming release.
+ *  Use `static_assert` for C++. For C, use either `_Static_assert` or
```

Comment:
I'd rephrase this to just

    Use `static_assert` instead. For C this is provided by `<assert.h>`, and for C++ it is a built-in keyword.

We'd never expect anyone to use `_Bool`, so no point telling them to use `_Static_assert`.

Reply:
Done.